tools/fuzzer/lldb_vs_lldb_eval_libfuzzer_test.cc

```diff
@@ -32,6 +34,40 @@
 
 static fuzzer::LibfuzzerState g_state;
 
+static const char* kLoggingPrefix = "[lldb-eval-fuzzer] ";
+
+// How often to log statistics on UB (e.g. every 500 iterations)?
+static const int kUbStatsPeriod = 500;
+
+extern "C" const char* __lsan_default_options() {
+  return "print_suppressions=0";
+}
+
+extern "C" const char* __lsan_default_suppressions() {
+  return "leak:printNode(llvm::itanium_demangle::Node const*\n"
+         "leak:llvm::itanium_demangle::initializeOutputStream";
+}
+
+class UbStats {
+ public:
+  void process(lldb_eval::UbStatus status) {
+    total_++;
+    count_[status]++;
+  }
+
+  bool should_report() const {
+    return total_ > 0 && total_ % kUbStatsPeriod == 0;
+  }
+
+  size_t count(lldb_eval::UbStatus status) { return count_[status]; }
+
+ private:
+  size_t total_;
+  std::unordered_map<lldb_eval::UbStatus, size_t> count_;
+};
+
```

Comment:
This is good approach if things were more complicated, but for now I would opt in for simplicity :0

```
void report_undefined_behaviour(lldb_eval::UbStatus status) {
  static std::unordered_map<lldb_eval::UbStatus, size_t> stats;
  static size_t count;

  stats[status]++;
  count++;

  if (count % kUbStatsPeriod == 0) {
    // inline log_ub_stats here
  }
}
```